### PR TITLE
[CELEBORN-1637] Enhance config to bypass memory check for partition file sorter

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
Add a config to bypass memory check when sorting shuffle files.

### Why are the changes needed?
If a celeborn worker has quite a large memory and it supports both Spark and Flink engines. This config should be enabled.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
Cluster test.

### CODE REVIEW

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Comment by s0nskar** [comment]
q: Wouldn't this be same as setting `directMemoryRatioThreshold` to 1?

**Comment by SteNicholas** [comment]
> directMemoryRatioThreshold

@s0nskar, this skips the memory check to continue partition sorting if a celeborn worker has quite a large memory, no need to wait memory ready.

**Comment by SteNicholas** [comment]
> q: Wouldn't this be same as setting `directMemoryRatioThreshold` to 1?

@s0nskar, this config option is only used to skip the memory check to continue partition sorting if a celeborn worker has quite a large memory, no need to wait memory ready.

**Comment by FMX** [comment]
> directMemoryRatioThreshold

```java
 public boolean sortMemoryReady() {
    return currentServingState() == ServingState.NONE_PAUSED
        && sortMemoryCounter.get() < maxSortMemory;
  }
```

No, it's different. The purpose of this config is to bypass the check of ServingState.  This PR can solve a problem when a Celeborn cluster is used for both Spark and Flink. The memory manager needs evolvements which I'll come up with a CIP later.

**Comment by s0nskar** [comment]
Got it, thanks for the explanation @FMX @SteNicholas

File: docs/configuration/worker.md [hunk 1]
```diff
@@ -161,6 +161,7 @@ license: |
 | celeborn.worker.shuffle.partitionSplit.min | 1m | false | Min size for a partition to split | 0.3.0 | celeborn.shuffle.partitionSplit.min | 
 | celeborn.worker.sortPartition.indexCache.expire | 180s | false | PartitionSorter's cache item expire time. | 0.4.0 |  | 
 | celeborn.worker.sortPartition.indexCache.maxWeight | 100000 | false | PartitionSorter's cache max weight for index buffer. | 0.4.0 |  | 
+| celeborn.worker.sortPartition.memoryCheck.bypass | false | false | Bypass memory check in partition files sorter. If your worker has quite large direct memory and same Celeborn cluster supports Spark and Flink, enable this setting will be helpful. | 0.6.0 |  |
```
Comment (pan3793) [inline docs/configuration/worker.md:164]: docs of `celeborn.worker.partitionSorter.directMemoryRatioThreshold` should be updated too.

Comment (FMX) [reply docs/configuration/worker.md:164]: Sounds good. I'll change this PR.

**Comment by pan3793** [comment]
> The memory manager needs evolvements which I'll come up with a CIP later.

I just saw this comment. For this case, I think we should classify configurations to each impl, for example

```
celeborn.worker.partitionSorter.memoryManager=none|abc|xyz
celeborn.worker.partitionSorter.abc.directMemoryRatioThreshold
celeborn.worker.partitionSorter.abc.otherProperty
celeborn.worker.partitionSorter.xyz.property1
celeborn.worker.partitionSorter.xyz.property2
```

then `celeborn.worker.partitionSorter.memoryManager=none` means disabling memory check. please correct me if I understand your proposal wrong.

**Review by pan3793 — APPROVED** [review]
lgtm except for a minor space issue.

**Comment by FMX** [comment]
> > The memory manager needs evolvements which I'll come up with a CIP later.
> 
> I just saw this comment. For this case, I think we should classify configurations to each impl, for example
> 
> ```
> celeborn.worker.partitionSorter.memoryManager=none|abc|xyz
> celeborn.worker.partitionSorter.abc.directMemoryRatioThreshold
> celeborn.worker.partitionSorter.abc.otherProperty
> celeborn.worker.partitionSorter.xyz.property1
> celeborn.worker.partitionSorter.xyz.property2
> ```
> 
> then `celeborn.worker.partitionSorter.memoryManager=none` means disabling memory check. please correct me if I understand your proposal wrong.

No, I mean that I want to track detailed memory usage. Evolving the memory to support fine-grade memory management, I think Flink's memory management can be some kind of reference, but skip its credit-based mechanism.

**Review by RexXiong — APPROVED** [review]

**Pull request closed without merging**